### DISCUSSION TITLE
How to prevent duplicate components in the editor?

### DISCUSSION THREAD

**Discussion by joaoaugustojr** [GrapesJS/grapesjs#5412] [Q&A] [answered]
I need to know how to prevent duplicate components in the editor, for example: I cannot allow two navBar components to exist in the editor, I create a script that removes this second one but only after adding it to the editor, would there be any way to do this before to drop the component in the editor? Thank you in advance for your help!!

**Comment by artf** [comment, marked as answer]
You can define `draggable` property as a function and use your own logic (eg. https://github.com/artf/grapesjs/discussions/4507)
